src/App.js

```diff
@@ -4,26 +4,41 @@ import './App.css';
 import chatMessages from './data/messages.json';
 import ChatLog from './components/ChatLog';
 
-const App = () => {
-  const [likes, setLikes] = useState(0);
-  const incrementLikes = () => {
-    setLikes(likes + 1);
+const countLikes = (messages) => {
+  let count = 0;
+  for (const message of messages) {
+    if (message.liked) {
+      count += 1;
+    }
   }
-  const decrementLikes = () => {
-    setLikes(likes - 1);
+  return count;
+};
+
+const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+  const onLikeClicked = (id) => {
+    setMessages(prev => {
+      return prev.map(currentMessage => {
+        if (id === currentMessage.id) {
+          return {...currentMessage, liked: !currentMessage.liked}        
+        } else {
+          return currentMessage;
+        }
+      })
+    });
   }
 
+  const likes = countLikes(messages);
   return (
     <div id="App">
       <header>
         <h1>Chat between Vladimir and Estragon</h1>
-        <h2>{likes} ❤️s</h2>    
+        <h2>{likes} ❤️s</h2>
       </header>
       <main>
-          <ChatLog entries={chatMessages} incrementLikes={incrementLikes} decrementLikes={decrementLikes} />
+          <ChatLog entries={messages} onLikeClicked={onLikeClicked}/>
```

Comment:
👍 Now we're passing the state-managed list of messages rather than the data we loaded directly from the JSON file.